### PR TITLE
Install, upgrade or rollback helm chart releases on a kubernetes cluster

### PR DESCRIPTION
**Pull request by tomasvalkering**

### CODE REVIEW

File: .github/workflows/helm_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        inputs:
+            ref_name:
+                required: true
+                type: string
+                description: tag or branch name to get values.yml file from
+            release:
+                required: true
+                type: string
+                description: the name of the release e.g. ubauth-ubique
+            command:
```
[the remark this answers is not in the document]
Comment (tomasvalkering) [reply .github/workflows/helm_deploy.yml:13]: Makes sense, will change that

File: .github/workflows/helm_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        secrets:
+            az_service_principal:
+                required: false
+            az_client_id:
+                required: false
+            az_tenant_id:
+                required: false
+            az_subscription_id:
+                required: false
+            workflow_pat_github:
```
[the remark this answers is not in the document]
Comment (kistlers) [reply .github/workflows/helm_deploy.yml:63]: As you just demonstrated, one could not guess what "pat" stands for, so we should write it out. Long identifer are not a problem when it makes stuff so much clearer

[the remark this answers is not in the document]
Comment (martinalig) [reply .github/workflows/helm_deploy.yml:63]: I hät jetzt uf workflow_patrick_github tippt 😂

File: .github/workflows/helm_deploy.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    contents: read
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        steps:
+            -   uses: actions/checkout@v4
+                with:
+                    ref: ${{ inputs.ref_name }}
+
```
[the remark this answers is not in the document]
Comment (kistlers) [reply .github/workflows/helm_deploy.yml:77]: I see. But then, this workflow is only for non-releases.helm charts from a repo.

Either, this should be reflected in the workflow name and we have a second workflow that uses a release workflow (that does not need the checkout)

Or, we make the checkout optional by some binary input and either apply the checked out helm chart or the released one.

**Comment by UBaggeler** [comment]
+1 for the personal access token.

Apart from that, I am wondering, whether we want to ask for all these arguments to be entered at each run. On one hand I fear, it is prone to make errors when entering the wrong combination of arguments but also it is probably hard to have a log of all the runs and changes. 

Are perhaps I am just not grasping the full extent of this: Are we storing the arguments / deployments in a separate repo and then we're just changing values in there? This would then be very similar to what we are doing with our Ansible / Helm Actions, which describe the expected state and in the history we can always see what changed at what time.

**Comment by tomasvalkering** [comment]
Added (untested) support for packaged charts as oci artifacts or tgz files. With oci on azure we should be able to avoid using the access token.

**Review by kistlers — COMMENTED** [review]
The workflows itself look good to me. But, I cannot really comment on the use-case for that (faggelers comment) or the correctness of the helm command/helm stuff. That is for others to check or decide :)

File: .github/workflows/helm_deploy.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ubamrein) [reply .github/workflows/helm_deploy.yml]: if the secret in the secret store literally is called like that it would only lead to confusing problems

**Review by kistlers — APPROVED** [review]

**Pull request merged**